src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> afterAdding = getNode(index);
+        Node<T> beforeAdding = afterAdding.prev;
+        Node<T> newNode = new Node<>(beforeAdding, value, afterAdding);
+        afterAdding.prev = newNode;
+        if (beforeAdding != null) {
+            beforeAdding.next = newNode;
+        } else {
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> elementToEdit = getNode(index);
+        T oldElement = elementToEdit.item;
+        elementToEdit.item = value;
+        return oldElement;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> elementToRemove = getNode(index);
+        unlink(elementToRemove);
+        return elementToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentValue = head;
+        for (int i = 0; i < size; i++) {
+            if (isElementsEqual(currentValue.item, object)) {
+                unlink(currentValue);
+                return true;
+            }
+            currentValue = currentValue.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNode(int index) {
+        indexCheckForGet(index);
+        Node<T> currentValue;
+        if ((size / 2) > index) {
+            currentValue = head;
+            for (int i = 0; i < index; i++) {
+                currentValue = currentValue.next;
+            }
+        } else {
+            currentValue = tail;
+            for (int i = size - 1; i > index; i--) {
+                currentValue = currentValue.prev;
+            }
+        }
+        return currentValue;
+    }
+
+    private void unlink(Node<T> node) {
+        size--;
+        if (node.next == null) {
+            if (node.prev != null) {
+                node.prev.next = null;
+            }
+            tail = node.prev;
+        } else if (node.prev == null) {
+            node.next.prev = null;
+            head = node.next;
+        } else {
+            node.next.prev = node.prev;
+            node.prev.next = node.next;
+        }
```

Comment:
let's see 1 more time if all cases are covered
if it is the last node (both head and tail)? if tail? if head? if neither head nor tail?